pkg/activator/net/throttler.go

```diff
@@ -306,8 +311,19 @@ func assignSlice(trackers []*podIPTracker, selfIndex, numActivators int) []*podI
 	sort.Slice(trackers, func(i, j int) bool {
 		return trackers[i].dest < trackers[j].dest
 	})
-	bi, ei := pickIndices(lt, selfIndex, numActivators)
-	return trackers[bi:ei]
+	bi, ei, remnants := pickIndices(lt, selfIndex, numActivators)
+	x := append(trackers[:0:0], trackers[bi:ei]...)
+	if remnants > 0 {
+		// This is basically: x = append(x, trackers[len(trackers)-remnants:]...)
+		// But we need to update the capacity.
+		for i := len(trackers) - remnants; i < len(trackers); i++ {
+			t := trackers[i]
+			// minOneOrValue ensures that infinity tracker will never scale to 0.
+			t.b.UpdateConcurrency(minOneOrValue(cc / numActivators))
```

Comment:
As discussed offline: theoretically if the code changes it might. But as written it is impossible.
But I fixed it anyway, since better code is better :)